test/accelerators/NNPA/numerical/TestGRU-NNPA.cpp

```diff
@@ -0,0 +1,114 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+//=========-- TestGRU-NNPA.cpp - Test configurations for GRU Models -=========//
+//
+// Copyright 2019-2022 The IBM Research Authors.
+//
+// =============================================================================
+//
+// This file contains configurations for testing GRU models on NNPA.
+// This file is based on test/numerical/TestGRU.cpp, but includes NNPA-specific
+// configurations. For example, bidirectional GRU is not included because it is
+// not supported in onnx-mlir. LinierBeforeReset is disabled because the zDNN
+// library used in NNPA does not support it.
+//
+//===----------------------------------------------------------------------===//
+
+#include <rapidcheck.h>
+
+#include "llvm/Support/FileSystem.h"
+
+#include "include/OnnxMlirRuntime.h"
+#include "src/Runtime/OMTensorHelper.hpp"
+#include "test/modellib/ModelLib.hpp"
+
+static const llvm::StringRef SHARED_LIB_BASE("./TestGRUNNPA_main_graph");
+
+using namespace mlir;
+
+namespace onnx_mlir {
+namespace test {
+
+// Returns whether onnx-mlir compiled GRU is producing the same results as a
+// naive implementation of GRU for a specific set of GRU
+// parameters/configuration.
+bool isOMGRUTheSameAsNaiveImplFor(const int direction, const int S, const int B,
+    const int I, const int H, const int linearBeforeReset,
+    bool isDynamicS = false, bool isDynamicB = false) {
+
+  GRULibBuilder gru(SHARED_LIB_BASE.str(), direction, S, B, I, H,
+      linearBeforeReset, isDynamicS, isDynamicB);
+  return gru.build() && gru.compileAndLoad() && gru.prepareInputs() &&
+         gru.run() && gru.verifyOutputs();
+}
+
+} // namespace test
+} // namespace onnx_mlir
+
+int main(int argc, char *argv[]) {
+  using namespace onnx_mlir;
+  using namespace onnx_mlir::test;
+
+  llvm::FileRemover remover(
+      ModelLibBuilder::getSharedLibName(SHARED_LIB_BASE.str()));
+
+  ModelLibBuilder::setRandomNumberGeneratorSeed("TEST_SEED");
+  setCompilerOptions({{OptionKind::CompilerOptLevel, "3"}});
+  llvm::cl::ParseCommandLineOptions(
+      argc, argv, "TestGRU\n", nullptr, "TEST_ARGS");
+
+  // RapidCheck test case generation.
+  bool success = rc::check("GRU implementation correctness", []() {
+    // The number of directions.
+    // 1: forward, -1: reverse, 2: bidirectional
+    // const auto D = *rc::gen::element(1, -1, 2);
+    // bidirectional not supported now
+    const auto D = *rc::gen::element(1, -1, 2);
```

Comment:
If that was the only difference, maybe we could create a TEST_RNN_MODES=full or reduced, and use the same main as in the normal test but setting this env var to different value on the CmakeFiles.txt ?

Reply:
@imaihal let me know when the changes are in, thanks

Reply:
@AlexandreEichenberger Done. I inserted some ifdef directives in TestLSTM, TestGemm, TestMatMul, and TestGRU, and they are set in `test/accelerators/NNPA/numerical/CMakeLists.txt` So, now only CMakeLists.txt is in the directory.